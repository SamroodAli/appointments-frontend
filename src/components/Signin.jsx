import { useState } from 'react';
import rails from '../api/rails';

const Signin = () => {
  const [email, setEmail] = useState('samrood@gmailexample22.com');
  const [password, setPassword] = useState('password');

  const onSubmit = async (e) => {
    e.preventDefault();
    const { data } = await rails.post('/auth/signin', {
      auth: {
        email,
        password,
      },
    }, {
    });

    console.log(data);
  };

  return (
    <form onSubmit={onSubmit}>
      <label htmlFor="username">
        Enter email
        <input value={email} onChange={(e) => setEmail(e.target.value)} />
      </label>
      <label htmlFor="username">
        Enter password
        <input value={password} onChange={(e) => setPassword(e.target.value)} />
      </label>
      <input type="submit" value="Submit" />
    </form>
  );
};

export default Signin;
