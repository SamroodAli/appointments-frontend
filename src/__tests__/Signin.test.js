import renderer from 'react-test-renderer';
// import { render } from '@testing-library/react';
import Signin from '../components/Signin';
import Wrapper from '../testWrapper';

test('Signin snapshot testing', () => {
  const tree = renderer.create(
    <Wrapper>
      <Signin />
    </Wrapper>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
