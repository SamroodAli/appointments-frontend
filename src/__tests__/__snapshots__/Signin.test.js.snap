// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin snapshot testing 1`] = `
<div
  className="flex justify-center items-center h-full bg-secondary"
>
  <div
    className="w-full max-w-xs "
  >
    <form
      className="bg-white shadow-md rounded px-8 pt-6 pb-8 mb-4"
      onSubmit={[Function]}
    >
      <div
        className="mb-4"
      >
        <label
          className="block text-gray-700 text-sm font-bold mb-2"
          htmlFor="username"
        >
          Email
          <input
            className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
            id="username"
            onChange={[Function]}
            placeholder="Username"
            type="text"
            value="samrood@gmailexample22.com"
          />
        </label>
      </div>
      <div
        className="mb-6"
      >
        <label
          className="block text-gray-700 text-sm font-bold mb-2"
          htmlFor="password"
        >
          Password
          <input
            className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 mb-3 leading-tight focus:outline-none focus:shadow-outline"
            id="password"
            onChange={[Function]}
            placeholder="******************"
            type="password"
            value="password"
          />
        </label>
      </div>
      <div
        className="flex items-center justify-around"
      >
        <input
          className="bg-green-300 hover:bg-primary text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
          type="submit"
          value="Sign in"
        />
        <a
          className="bg-blue-300 hover:bg-primary text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
          href="/signup"
          onClick={[Function]}
          value="Sign in"
        >
          Sign up ?
        </a>
      </div>
    </form>
  </div>
</div>
`;
